benchmarks/src/main/scala/zio/chunks/ArrayBenchmarks.scala

```diff
@@ -1,55 +1,63 @@
-package zio
+package zio.chunks
 
 import java.util.concurrent.TimeUnit
 
 import org.openjdk.jmh.annotations._
+import zio.Chunk
 @State(Scope.Thread)
 @BenchmarkMode(Array(Mode.AverageTime))
 @OutputTimeUnit(TimeUnit.NANOSECONDS)
-class ChunkBenchmarks {
+class ArrayBenchmarks {
 
   @Param(Array("100", "1000", "10000"))
-  var chunkSize: Int = _
+  var array: Array[Int] = _
 
   var chunk: Chunk[Int] = _
 
-  var array: Array[Int] = _
+  var chunkSize: Int = _
 
   @Setup(Level.Trial)
   def setup() = {
     array = Array.fill(0)(chunkSize)
     chunk = Chunk.fromArray(array)
   }
 
-  @Benchmark
-  def chunkFold(): Int =
-    chunk.fold(0)(_ + _)
-
   @Benchmark
   def arrayFold(): Int =
     array.sum
-  @Benchmark
-  def chunkMap(): Chunk[Int] = chunk.map(_ * 2)
 
   @Benchmark
   def arrayMap(): Array[Int] = array.map(_ * 2)
 
   @Benchmark
-  def chunkFlatMap(): Chunk[Int] = chunk.flatMap(n => Chunk(n + 2))
+  def arrayMapOptimized(): Array[Int] = {
+    var i   = 0
+    val len = array.length
 
-  @Benchmark
-  def arrayFlatMap(): Array[Int] = array.flatMap(n => Array(n + 2))
+    while (i < len) {
+      array(i) = array(i) * 2
```

Comment:
We should give it a completely "fair" treatment, and not mutate the original array. Just allocate another array and initialize it with mapping results.